javascriptv3/example_code/bedrock-agent-runtime/actions/invoke-flow.js

```diff
@@ -0,0 +1,73 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { fileURLToPath } from "node:url";
+
+import {
+  BedrockAgentRuntimeClient,
+  InvokeFlowCommand,
+} from "@aws-sdk/client-bedrock-agent-runtime";
+
+/**
+ * Invokes an alias of a flow to run the inputs that you specify and return
+ * the output of each node as a stream.
+ *
+ * @param {string} flowIdentifier - The unique identifier of the flow.
+ * @param {string} flowAliasIdentifier - The unique identifier of the flow alias.
+ * @param {string} prompt - The input to send to the prompt flow input node.
+ * @param {string} [region='us-east-1'] - The AWS region in use.
+ * @returns {Promise<import("@aws-sdk/client-bedrock-agent").FlowNodeOutput>} An object containing information about the output from flow invocation.
+ */
+export const invokeBedrockFlow = async (
+  flowIdentifier,
+  flowAliasIdentifier,
+  prompt,
+  region = "us-east-1",
+) => {
+  const client = new BedrockAgentRuntimeClient({ region });
+
+  const command = new InvokeFlowCommand({
+    flowIdentifier,
+    flowAliasIdentifier,
+    inputs: [
+      {
+        content: {
+          document: prompt,
+        },
+        nodeName: "FlowInputNode",
+        nodeOutputName: "document",
+      },
+    ],
+  });
+
+  let flowResponse = {};
+  const response = await client.send(command);
+
+  for await (const chunkEvent of response.responseStream) {
+    const { flowOutputEvent, flowCompletionEvent } = chunkEvent;
+
+    if (flowOutputEvent) {
+      flowResponse = { ...flowResponse, ...flowOutputEvent };
+      console.log("Flow output event:", flowOutputEvent);
+    } else if (flowCompletionEvent) {
+      flowResponse = { ...flowResponse, ...flowCompletionEvent };
+      console.log("Flow completion event:", flowCompletionEvent);
+    }
+  }
+
+  return flowResponse;
+};
+
+// Call function if run directly
+if (process.argv[1] === fileURLToPath(import.meta.url)) {
+  const flowIdentifier = "[YOUR_FLOW_ID]";
```

Comment:
Hey Dennis, we updated our standards to disallow hardcoded strings like this.
https://github.com/awsdocs/aws-doc-sdk-examples/wiki/general-code-example-standards#guidelines-5
The gist is that code examples should not require a customer to edit the code to run the example. Can you modify this to take user input for the variables?

Reply:
I pushed an update for this.